src/editor/components/menu/editor-mode-title.jsx

```diff
@@ -61,7 +62,10 @@ export function mapStateToProps(state) {
 
 export function mapDispatchToProps(dispatch) {
   return {
-    changePageTitle: title => dispatch(changePageTitle(title))
+    updateTitle: title => {
+      dispatch(updateTitle(title));
+      dispatch(updateAutosave());
```

Comment:
as mentioned elsewhere: i won't block merge on this, just want to flag if as something to look at

-BUT- please do add a FIXME mentioning this seperation-of-concerns question

Reply:
Maybe we need to somehow be able to distinguish operations which modify the title/content based on a user action, vs. ones that are based on something else (e.g. the document being loaded for the first time).

Reply:
yuck, document initialization is a whole badness unto itself that needs a lot of work. i see what you mean -- it looks like that action creator is only ever dispatched during this user action or during initialization. i think your suggestion of changing the semantics of that bool arg passed into update title to something like `updateTitle(title, userAction=true)`  makes a lot of sense. i support that if that's the way you want to go, but again: i won't block on that for this PR if you just want to add a FIXME here instead

Reply:
yes i think that sounds right. To elaborate on that idea, maybe something like: move `updateJsmdContent` and `updateTitle` to e.g. "update-content-actions.js", and wrap them in `userUpdateJsmdContent` and `userUpdateTitle`, which can also call the autosaving functions.

looking at it briefly, i even think that by moving these functions out to their own files ("user-content-update-actions.js" and "content-update-actions.js") we _might_ even be able to untangle the last of the import cycles... it looks like this may be the crux of the last couple cycles

Reply:
I think I'd like to do this in a followup if you don't mind. I'll file an issue right now and link to this discussion

Reply:
See #1862